packages/twenty-front/src/modules/object-record/record-group/hooks/useCurrentRecordGroupId.ts

```diff
@@ -1,20 +1,37 @@
+/* eslint-disable no-redeclare */
+/* eslint-disable prefer-arrow/prefer-arrow-functions */
 import { RecordGroupContext } from '@/object-record/record-group/states/context/RecordGroupContext';
 import { useContext } from 'react';
+import { isDefined } from '~/utils/isDefined';
 
-export const useCurrentRecordGroupId = () => {
+type UseCurrentRecordGroupIdOptions = {
+  allowUndefined?: boolean;
+};
+
+export function useCurrentRecordGroupId(): string;
+export function useCurrentRecordGroupId(options?: {
+  allowUndefined: true;
+}): string | undefined;
+export function useCurrentRecordGroupId(options?: {
+  allowUndefined?: false;
+}): string;
+
+export function useCurrentRecordGroupId(
+  options?: UseCurrentRecordGroupIdOptions,
+): string | undefined {
   const context = useContext(RecordGroupContext);
 
-  if (!context) {
+  if (!context && !isDefined(options?.allowUndefined)) {
```

Comment:
General rule of thumb here : we should have the flow of execution rely on a ternary in the tree rendering part instead of TypeScript.

While it's possible to use TypeScript for that, it's always much simpler to know that we're in a part of the tree where currentRecordGroupId should be defined and another part where it is not used at all.